### DISCUSSION TITLE
Will Globe View be developed on MapLibre?

### DISCUSSION THREAD

**Discussion by jsteeew** [maplibre/maplibre-gl-js#1598] [Ideas]
It's a pretty sweet feature: https://www.mapbox.com/blog/globe-view

Would love to see it implemented!! :)

**Comment by HarelM** [comment]
Depends on how much the community is willing to push this forward, the issue is there for a long time waiting for someone to pick it up :-)
https://github.com/maplibre/maplibre-gl-js/issues/307

**Comment by klokan** [comment]
Yes - progress is visible here https://github.com/maplibre/maplibre/discussions/161

**Comment by jatech1** [comment]
Just curious on projected timeframe for globe release.  From the activity and examples it seemed like it was close, but the latest newsletter says by end of year.  I know there is a lot of effort going in to this, and it is very much appreciated!

**Comment by freeseus** [comment]
This is something of significant interest to me, and I intend on using a globe projection (in part), especially if it features a starfield, horizon, bump/normals mapping (for terrain elevation), a clouds layer, and other eye-candy-like features.

I have wanted to be able to do something like: https://www.fonterra.com/nz/en/campaign/fonterra-globe.html#/home and combine it with a tile-based map system like Maplibre gl for a fictional map for several years now.

In my project, I have made great progress using Maplibre gl with the flat mercator projection and included dynamic content in the form of GLSL shaders. Being able to leverage a globe projection with the shaders being properly displayed in the right places is something that I value greatly. In my spare time, I've spent roughly 2 years building a high-resolution custom fantasy map (chopped into tiles) with dynamic events and game mechanics. Being able to use the globe projection and transition back smoothly would be more than just icing on the cake.

I frequently check the link klokan posted above and the Maplibre gl releases pages to see progress updates and new features.

**Reply by HarelM** [reply]
There is a globe branch, it's almost complete, you can take it for a spin and see what's still missing. Feel free to contribute bug fixes and other improvements.

**Comment by willemarcel** [comment]
I've tested the globe view with some data that crosses the North Pole and it's rendered that way
<img width="829" alt="Screenshot 2024-08-30 at 13 47 26" src="https://github.com/user-attachments/assets/8f233095-95ee-43ec-bd13-11e9155af881">

I know that the web Mercator tiles don't cover that part of the planet, however do you think MapLibre would be able to render a vector data layer, such as a geojson file, over the poles correctly in the future?

**Reply by HarelM** [reply]
I would suggest to create a simplified version of the data to showcase the issue and post it in the following PR:
https://github.com/maplibre/maplibre-gl-js/pull/3963
Thanks for taking the time to test this out!
